kikuchipy/indexing/osm.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2020 The kikuchipy developers
+#
+# This file is part of kikuchipy.
+#
+# kikuchipy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kikuchipy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kikuchipy. If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Tuple, List
+from math import copysign
+
+import numpy as np
+from scipy.ndimage import generic_filter
+
+from orix.crystal_map import CrystalMap
+
+from kikuchipy.indexing.similarity_metrics import (
+    SimilarityMetric,
+    SIMILARITY_METRICS,
+)
+
+
+def orientation_similarity_map(
+    xmap: CrystalMap,
+    n_largest: int = None,
+    normalize_by_n: bool = True,
+    from_n_largest: int = None,
+    footprint: np.ndarray = None,
+    center_index: int = 2,
+) -> np.ndarray:
+    """Create Orientation Similarity Map(OSM), following [Marquardt2017]_.
+
+    The given CrystalMap `xmap` must contain `"template_indices"` in `xmap.prop`
+    as produced by :class:`~kikuchipy.indexing.StaticDictionaryIndexing`.
+
+    Parameters
+    ----------
+    xmap : CrystalMap
+        CrystalMap with "template_indices" in prop
+    n_largest : int, optional
+        Number of sorted results to be used, by default use all.
+    normalize_by_n : bool, optional
+        Whether to return Orientation Similarity in range [0, 1], by default True
+    from_n_largest : int, optional
+        Create OSM for each n in range [`from_n_largest`, `n_largest`], by default None.
+    footprint : [np.ndarray], optional
+        Boolean 2D array specifying which neighbours to be used in computation of OS
+        in each navigation point, by default nearest neighbours.
+    center_index : int, optional
+        Flat index of central navigation point in the truthy values of footprint, by default 2.
+
+    Returns
+    -------
+    osm : np.ndarray
+        Orientation Similarity Map(s)
+
+    Notes
+    -----
+    If a range of OSM are to be created n_largest is at osm[:,:,0]
+    and from_n_largest at osm[:,:,-1].
+
+    Raises
+    ------
+    NotImplementedError
+        [might be interesting to try different footprints]
+
+    References
+    ----------
+    .. [Marquardt2017] K. Marquardt, M. De Graef,\
+        S. Singh, H. Marquardt, A. Rosenthal, S. Koizuimi\
+        "Quantitative electron backscatter diffraction (EBSD) data analyses \
+        using the dictionary indexing (DI) approach"\
+        , American Mineralogist: Journal of Earth and Planetary Materials 102(9), 2017.
+    """
```

Comment:
It would be great to add a `Notes` part to the docstring with an equation explaining how the intersection is computed. See e.g. here for an example of how maths are written in Sphinx: https://github.com/pyxem/kikuchipy/blob/master/kikuchipy/filters/window.py#L506-L567

Reply:
This is now added.